### PR TITLE
pipeline: Add a new `test_gpdb_ent_debian_package` job

### PR DESCRIPTION
**Pull request by goutamtadi1**
- Add `libxml` & `quicklz` tests
- Refactor `test_gpdb_deb_package` to `test_gpdb_ent_debian_package` and
`test_gpdb_open_source_debian_package`
- Add `libsnmp-dev` for ent debian package

### CODE REVIEW

File: src/tools/docker/ubuntu16-debian/debian/control [hunk 1]
```diff
@@ -17,6 +17,7 @@ Depends: bison,
     libxerces-c3.1,
     libxml2-dev,
     locales,
+    libsnmp-dev,
```
Comment (kmacoskey) [inline src/tools/docker/ubuntu16-debian/debian/control]: Why is this necessary for adding a new `test_gpdb_ent_debian_package` job?

File: concourse/scripts/deb_test_package.bash [hunk 2]
```diff
@@ -7,8 +7,18 @@ apt-get install -y ./${DEBIAN_PACKAGE:-deb_package_ubuntu16/greenplum-db.deb}
...
 su - gpadmin ${CURRENT_DIR}/gpdb_src/concourse/scripts/deb_init_cluster.bash
-su - gpadmin ${CURRENT_DIR}/gpdb_src/concourse/scripts/deb_test_cluster.bash
 
-su - gpadmin -c "source /opt/gpdb/greenplum_path.sh && psql -t -U gpadmin template1 -c 'select version()' | grep '\-oss'"
-# gpssh replaces "-" with " " in version string
-su - gpadmin -c "source /opt/gpdb/greenplum_path.sh && gpssh --version | grep ' oss'"
+if ${IS_OPEN_SOURCE_DEB} ; then
+    su - gpadmin ${CURRENT_DIR}/gpdb_src/concourse/scripts/open_source_deb_test_cluster.bash
+else
+   su - gpadmin -c "source /opt/gpdb/greenplum_path.sh && psql -t -U gpadmin -v ON_ERROR_STOP=ON template1" << EOF
```
Comment (kmacoskey) [inline concourse/scripts/deb_test_package.bash]: `open_source_deb_test_cluster.bash` gets it's own, separate task script, why not do the same for what would be the `enterprise_deb_test_cluster.bash` ?  The inconsistency is confusing when the goals of each part of the conditional is the same, test if it's a valid OSS build or test if it's a valid Enterprise build. Why have one a separate script and not the other?

File: concourse/scripts/deb_test_package.bash [hunk 3]
```diff
@@ -7,8 +7,18 @@ apt-get install -y ./${DEBIAN_PACKAGE:-deb_package_ubuntu16/greenplum-db.deb}
 locale-gen en_US.UTF-8
 ${CURRENT_DIR}/gpdb_src/concourse/scripts/setup_gpadmin_user.bash
 su - gpadmin ${CURRENT_DIR}/gpdb_src/concourse/scripts/deb_init_cluster.bash
```
Comment (kmacoskey) [inline concourse/scripts/deb_test_package.bash]: Why is it necessary to have a separate process for initing a cluster? Is it not sufficient to re-use the same CI code that runs `make create-demo-cluster`? A lot of that exists in `concourse/scripts/common.bash`. There would need to be some small refactoring, but then we are only maintaining a single process for initing a cluster in CI for testing. Refactor and re-use before reproducing unless there are different problems that need to be solved that cannot be consolidated into the same CI code.

File: src/tools/docker/ubuntu16-debian/debian/control (hunk 1 above)
Comment (goutamtadi1) [reply src/tools/docker/ubuntu16-debian/debian/control]: @kmacoskey We're experiencing the following error without that library when initializing greenplum cluster
```
/opt/gpdb/bin/postgres: error while loading shared libraries: libnetsnmp.so.30: cannot open shared object file: No such file or directory
```

Comment (kmacoskey) [reply src/tools/docker/ubuntu16-debian/debian/control]: Yep, the enterprise compilation for ubuntu 16.04 is currently including configuration with snmp enabled:
https://github.com/greenplum-db/gpdb/blob/2e22db3c5409dc4d6474f6f1f86294d8900e7a0d/gpAux/Makefile#L177

Is snmp support necessary for the enterprise ubuntu 16.04 compilation?

 As a style suggestion, just based on opinion, a separate commit that adds this as a dependency based on the pre-existing decision that snmp will be included in the enterprise build would be useful. For all of the dependencies listed in `src/tools/docker/ubuntu16-debian/debian/control`, I think it is reasonable to assume that our future selves may want easy ways to find the context about why each dependency was included. This is an existing issue we are having to deal with for gpdb6, the lack of clear commits and context around dependencies.

Comment (goutamtadi1) [reply src/tools/docker/ubuntu16-debian/debian/control]: RHEL6 and RHEL7 platforms are also including snmp support in the `gpdb/gpAux/Makefile`. We think enterprise users expect this. We can validate this with @ozbasarir if you have different thoughts. We will make it a separate commit to include `libsnmp-dev`.

File: concourse/scripts/deb_test_package.bash (hunk 3 above)
Comment (goutamtadi1) [reply concourse/scripts/deb_test_package.bash]: ✅

**Comment by goutamtadi1** [comment]
Good point about the group of ICW. We have an upcoming chore to expand these tasks into doing ICW (we'll send link to chore in separate communique). So these tasks will soon be renamed to "icw.*" and actually do ICW.  

Is it ok to add these now, in light of this upcoming chore?

**Comment by kmacoskey** [comment]
Having ICW be ran against the enterprise ubuntu 16.04 build would be good - do you think it would be valuable to do that AND keep the separate ICW testing for the OSS ubuntu 16.04 build? If the differences between enterprise and OSS are just in a few feature flags during configuration and some manual actions to bring in compression libraries, then it may not be as useful to maintain both. 

That being said, I would actually suggest giving some thought about keeping these above tests separate from anything done with ICW. There could be some value in building the sense of basic tests to validate that you have correctly compiled for what we define as an enterprise or OSS build and maybe further then that, checking that you have built a valid deb package. Checking that libxml and quicklz are available are great examples of that.

> Is it ok to add these now, in light of this upcoming chore?

Indeed it is - I don't think ICW is even a necessary part of the value that can be made from the idea of these tests. Testing that an enterprise build inside a deb package is indeed an enterprise build is a different problem that doesn't need the ICW hammer thrown at it.

**Comment by jpatel-pivotal** [comment]
@kmacoskey Thanks for the perspective.

We plan to cancel this PR, in favor of the following:

* move the packaging validation to the gp4k pipeline (done)
* remove the current `test_gpdb_debian_package` job via new PR. 
* check for xml and quicklz  in other pipelines (already happening)
* remove snmp configure flag in PR: https://github.com/greenplum-db/gpdb/pull/6381


 
--Jemish & Larry
